pkg/skaffold/deploy/kustomize_test.go

```diff
@@ -90,10 +99,38 @@ func TestKustomizeDeploy(t *testing.T) {
 			},
 			forceDeploy: true,
 		},
+		{
+			description: "built-in kubectl kustomize",
+			kustomize: latest.KustomizeDeploy{
+				KustomizePaths: []string{"a", "b"},
+			},
+			commands: testutil.
+				CmdRunOut("kubectl version --client -ojson", kubectlVersion118).
+				AndRunOut("kubectl --context kubecontext --namespace testNamespace kustomize a", deploymentWebYAML).
+				AndRunOut("kubectl --context kubecontext --namespace testNamespace kustomize b", deploymentAppYAML).
+				AndRunInputOut("kubectl --context kubecontext --namespace testNamespace get -f - --ignore-not-found -ojson", deploymentWebYAMLv1+"\n---\n"+deploymentAppYAMLv1, "").
+				AndRun("kubectl --context kubecontext --namespace testNamespace apply -f - --force --grace-period=0"),
+			builds: []build.Artifact{
+				{
+					ImageName: "leeroy-web",
+					Tag:       "leeroy-web:v1",
+				},
+				{
+					ImageName: "leeroy-app",
+					Tag:       "leeroy-app:v1",
+				},
+			},
+			forceDeploy:        true,
+			kustomizeCmdAbsent: true,
+		},
 	}
+
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
 			t.Override(&util.DefaultExecCommand, test.commands)
+			if test.kustomizeCmdAbsent {
+				t.Override(&kustomizeBinaryCheck, kustomizeAbsent)
+			}
```

Comment:
You can rename testKustomizeCmdAbsent to testKustomizeCmdPresent and instead override the function directly here.
```suggestion
			t.Override(&kustomizeBinaryCheck, func () bool { return test.kustomizeCmdPresent} )
```